src/components/scenes/TransactionDetailsScene.tsx

```diff
@@ -310,82 +310,113 @@ export const TransactionDetailsComponent: React.FC<Props> = (props: Props) => {
+          (lowerCat === 'income:' && direction === 'receive') ||
+          (lowerCat === 'expense:' && direction === 'send')
+        ) {
+          nextCategory = savedData.category
+          categoryChange = null
+        } else {
+          nextCategory = incomingCategory
+          categoryChange = incomingCategory
+        }
       }
-    }
 
-    if (notes !== localMetadata.notes) {
-      changed = true
-      if (notes === savedData.notes || notes === '') {
-        newNotes = null
-        newDetails.notes = savedData.notes
-      } else {
-        newNotes = notes
+      // Notes
+      let nextNotes = prev.notes ?? ''
+      let notesChange: string | null | undefined
+      if (newDetails.notes !== undefined && newDetails.notes !== prev.notes) {
+        const incomingNotes = newDetails.notes
+        if (incomingNotes === savedData.notes || incomingNotes === '') {
+          nextNotes = savedData.notes ?? ''
+          notesChange = null
+        } else {
+          nextNotes = incomingNotes
+          notesChange = incomingNotes
+        }
       }
-    }
-
-    if (!matchJson(exchangeAmount, localMetadata.exchangeAmount)) {
-      changed = true
-      newExchangeAmount = exchangeAmount
-    }
 
-    if (!changed) {
-      console.log('EXIT onSaveTxDetails no change')
-      return
-    }
-    const metadata: EdgeMetadataChange = {
-      name: newName,
-      category: newCategory,
-      notes: newNotes,
-      exchangeAmount: newExchangeAmount
-    }
+      const anyChanged =
+        nextName !== (prev.name ?? '') ||
+        nextNotes !== (prev.notes ?? '') ||
+        nextCategory !== prev.category ||
+        !matchJson(mergedExchangeAmount, prev.exchangeAmount)
 
-    const saveTxMetadataParams: EdgeSaveTxMetadataOptions = {
-      txid: transaction.txid,
-      tokenId: transaction.tokenId,
-      metadata
-    }
+      if (!anyChanged) {
+        console.log('EXIT onSaveTxDetails no change')
+        return prev
+      }
 
-    wallet.saveTxMetadata(saveTxMetadataParams).catch(showError)
+      const exchangeAmountChange = matchJson(
+        mergedExchangeAmount,
+        prev.exchangeAmount
+      )
+        ? undefined
+        : mergedExchangeAmount
+
+      metadataToSave = {
+        name: nameChange,
+        category: categoryChange,
+        notes: notesChange,
+        exchangeAmount: exchangeAmountChange
+      }
 
-    setLocalMetadata(newValues)
+      const nextLocal: EdgeMetadata = {
+        exchangeAmount: mergedExchangeAmount,
+        bizId: prev.bizId ?? 0,
+        category: nextCategory,
+        name: nextName,
+        notes: nextNotes
+      }
+      return nextLocal
+    })
+
+    if (metadataToSave != null) {
+      const saveTxMetadataParams: EdgeSaveTxMetadataOptions = {
+        txid: transaction.txid,
+        tokenId: transaction.tokenId,
+        metadata: metadataToSave
+      }
+      wallet.saveTxMetadata(saveTxMetadataParams).catch(showError)
```

Comment:
This is a bad old pattern that should be cleaned up. Just put the parameters inline:

```ts
await wallet.saveTxMetadata({ 
  txid: ... ,
  tokenId: ...,
  metadata: ...
}).catch((error: unknown) => {
  showError(error)
})
```

Also note the `await`. This whole thing should be `async`, so we get the proper spinners & double-click prevention.